qlty-config/src/config/triage.rs

```diff
@@ -0,0 +1,134 @@
+use super::ignore::is_rule_issue_match;
+use super::IssueMode;
+use crate::config::issue_transformer::IssueTransformer;
+use globset::{Glob, GlobSet, GlobSetBuilder};
+use qlty_types::category_from_str;
+use qlty_types::{analysis::v1::Issue, level_from_str};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::sync::RwLock;
+
+#[derive(Debug, Serialize, Deserialize, Default, JsonSchema)]
+pub struct Match {
+    #[serde(default)]
+    pub plugins: Vec<String>,
+
+    #[serde(default)]
+    pub rules: Vec<String>,
+
+    #[serde(default)]
+    pub file_patterns: Vec<String>,
+
+    #[serde(skip)]
+    pub glob_set: RwLock<Option<GlobSet>>,
+}
+
+impl Clone for Match {
+    fn clone(&self) -> Self {
+        Self {
+            plugins: self.plugins.clone(),
+            rules: self.rules.clone(),
+            file_patterns: self.file_patterns.clone(),
+            glob_set: RwLock::new(None),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Default, Clone, JsonSchema)]
+pub struct Set {
+    #[serde(default)]
+    pub level: Option<String>,
+
+    #[serde(default)]
+    pub category: Option<String>,
+
+    #[serde(default)]
+    pub mode: Option<IssueMode>,
```

Comment:
I think `disabled` is allowed as an `IssueMode` but should be rejected here. So I guess what we currently call "IssueMode" is really a _plugin_ mode and an Issue Mode is a plugin mode _except_ "disabled"